File: GXpert/GXpert.Web/Modules/ServerTypes/Content/ContentMediaRatingColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ContentMediaRatingRow } from "./ContentMediaRatingRow";

export interface ContentMediaRatingColumns {
    Id: Column<ContentMediaRatingRow>;
    ContentTitle: Column<ContentMediaRatingRow>;
    RatingName: Column<ContentMediaRatingRow>;
    Score: Column<ContentMediaRatingRow>;
    Username: Column<ContentMediaRatingRow>;
}

export class ContentMediaRatingColumns extends ColumnsBase<ContentMediaRatingRow> {
    static readonly columnsKey = 'Content.ContentMediaRating';
    static readonly Fields = fieldsProxy<ContentMediaRatingColumns>();
}

File: GXpert/GXpert.Web/Modules/Content/ContentMediaRating/ContentMediaRatingColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Content.Columns;

[ColumnsScript("Content.ContentMediaRating")]
[BasedOnRow(typeof(ContentMediaRatingRow), CheckNames = true)]
public class ContentMediaRatingColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ContentTitle { get; set; }
    public string RatingName { get; set; }
    public short Score { get; set; }
    public string Username { get; set; }
  
}